### ISSUE TITLE
Time Remaining estimate is incorrect if END_DATE is before BEG_DATE+JOB_SGMT

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#1379]
Today I ran a 3-month C12 job from 20000414 21z to 20000802 21z which is 110 days. But I decided to be lazy and in CAP.rc set:
```
END_DATE:     20000802 210000
JOB_SGMT:     00001000 000000
NUM_SGMT:     1
```
figuring I'd like MAPL take the wheel and stop the model. I think this means I told it to run for 10 months (or 1000 days??). 

And all worked just as you'd expect; the model stopped at the `END_DATE`. But, the instant timers had a bit of an issue. The last report before the run ended was:
```
 AGCM Date: 2000/08/02  Time: 21:00:00  Throughput(days/day)[Avg Tot Run]:   1895.8   1688.9   1694.0  TimeRemaining(Est) 002:28:52   16.9% :  20.7% Mem Comm:Used
```

Obviously, there weren't 2h28m left, there was about 1 minute left.

The issue I think is that this estimate is based on `nsteps` and in Cap Grid Comp:

https://github.com/GEOS-ESM/MAPL/blob/0dbe0f14adf059e53d29112d0ee2da9f4d560275/gridcomps/Cap/MAPL_CapGridComp.F90#L1835

and we only ever calculate that based on duration (which is from job_sgmt). We should probably use the minimum of end vs (curr+sgmt) or something.

**Comment by atrayano** [comment]
BEG_DATE is not relevant, as long it is before curr. I am Time (and END_DATE). We need to limit 'duration' to the smaller of the user specified duration and (END_DATE-currTime). My fix will be in shortly (after running few simple tests)

**Comment by mathomp4** [comment]
Fixed with #1381
